### PR TITLE
Development to master branch

### PR DESCRIPTION
**Pull request by morganblackwell**
Code to push to master before it is reviewed

### CODE REVIEW

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -13,11 +13,26 @@ internal class Die
...
 
-        //Property
+        /// <summary> returns the currentValue of a randomly rolled 6 sided die. </summary>
 
+        // Private variable
+        private int _currentValue;
 
-        //Method
+        // Property
+        public int CurrentValue
```
Comment (Xeroas) [inline CMP1903_A1_2324/Die.cs:25]: More commenting could be done here to explain what is happening

File: CMP1903_A1_2324/Program.cs [hunk 2]
```diff
@@ -14,6 +14,29 @@ static void Main(string[] args)
...
+            /// Infinitely creates a game object through the Testing class
+            /// until the users types 'stop'.
+            /// </summary>
+
+
+            // Keep rolling until 'stop' is typed
+            for (; ; )
+            {
+                Testing testOutputs = new Testing();
+                testOutputs.Main();
```
Comment (Xeroas) [inline CMP1903_A1_2324/Program.cs]: Game seems to be ran through the testing class, not game class - is this intentional?

File: CMP1903_A1_2324/Testing.cs [hunk 3]
```diff
@@ -15,6 +16,39 @@ internal class Testing
...
          */
 
+        /// <summary>
+        ///  The testing class rolls 3 dice and checks if the following is true:
+        ///  Every roll is between the values 1 and 6
+        ///  The sum of rolls is between the values 3 and 18
+        /// </summary>
+
         //Method
+        public void Main()
```
Comment (Xeroas) [inline CMP1903_A1_2324/Testing.cs:26]: Good handling of all potential errors, case type is consistent with C# convention.

File: CMP1903_A1_2324/Testing.cs [hunk 4]
```diff
@@ -15,6 +16,39 @@ internal class Testing
...
+            Game testGame = new Game();
+            var testOutputs = testGame.Main();
+
+            int roll1 = testOutputs.Item1;
+            int roll2 = testOutputs.Item2;
+            int roll3 = testOutputs.Item3;
+            int sum = testOutputs.Item4;
+
+            // First dice testing
+            Debug.Assert(roll1 <= 6, "First dice roll is greater than 6");
```
Comment (Xeroas) [inline CMP1903_A1_2324/Testing.cs:37]: This could be further improved by running on one line using "&&"

File: CMP1903_A1_2324/Program.cs [hunk 5]
```diff
@@ -14,6 +14,29 @@ static void Main(string[] args)
...
              */
+
+            /// <summary>
+            /// Infinitely creates a game object through the Testing class
+            /// until the users types 'stop'.
+            /// </summary>
+
+
+            // Keep rolling until 'stop' is typed
+            for (; ; )
```
Comment (Xeroas) [inline CMP1903_A1_2324/Program.cs]: No Syntax errors, code runs as expected

File: CMP1903_A1_2324/Game.cs [hunk 6]
```diff
@@ -15,7 +15,30 @@ internal class Game
...
-        //Methods
+        /// <summary>  Creates three die objects, rolls them and outputs the values and their sum. </summary>
+        /// 
+        /// <returns> first die value, second die value, third die value, sum of all dice. </returns>
+
+        //Method
+        public (int, int, int, int) Main()
+        {
+            // Create 3 dice
+            Die die1 = new Die();
```
Comment (Xeroas) [inline CMP1903_A1_2324/Game.cs:27]: Formatted very nicely, very easy to read and understand what is happening

**Review by Xeroas — COMMENTED** [review]
Great piece of code, very little that could be improved!

1. Have the requirements been met? Yes
2. Is the code formatted using the Style Guidelines correctly? Yes
3. Is the code easy to read? Yes
4. Are different errors handled correctly? Die rolls are checked to ensure they are within the expected range but no formal (Try block) error handling is implemented, however the code handles unforeseen inputs in the one user input without any faults or breaking.

File: .idea/vcs.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
Comment (Tr0-p) [inline .idea/vcs.xml:6]: These files inside of inspectionProfiles can be added to Git Ignore.

File: CMP1903_A1_2324/Die.cs [hunk 8]
```diff
@@ -13,11 +13,26 @@ internal class Die
          * and one method that rolls the die, returns and integer and takes no parameters.
          */
 
-        //Property
+        /// <summary> returns the currentValue of a randomly rolled 6 sided die. </summary>
```
Comment (Tr0-p) [inline CMP1903_A1_2324/Die.cs]: Good use of XML commenting however some more detail could be added.

File: CMP1903_A1_2324/Die.cs [hunk 9]
```diff
@@ -13,11 +13,26 @@ internal class Die
...
 
+        // Private variable
```
Comment (Tr0-p) [inline CMP1903_A1_2324/Die.cs]: Redundant Comment. More detail can be added to show what is being stored.

File: CMP1903_A1_2324/Die.cs [hunk 10]
```diff
@@ -13,11 +13,26 @@ internal class Die
...
+            get { return _currentValue; }
+            set { _currentValue = value; }
+        }
 
+        private static Random _random = new Random(); // Using static stops duplicate values
 
+        // Method
+        // Returns a random integer between 1 and 6
+        public int Roll()
+        {
```
Comment (Tr0-p) [inline CMP1903_A1_2324/Die.cs:35]: Some Function XML comments can be added to describe what is going on in the function alongside expected returns.

File: CMP1903_A1_2324/Game.cs [hunk 11]
```diff
@@ -15,7 +15,30 @@ internal class Game
...
+            int sum = roll1 + roll2 + roll3; // Calculates the total value of all three dice rolls
+
+            Console.WriteLine($"Roll 1: {roll1}\nRoll 2: {roll2}\nRoll 3: {roll3}\nTotal: {sum}");
+
+            return (roll1, roll2, roll3, sum);
+        }
+
 
     }
 }
```
Comment (Tr0-p) [inline CMP1903_A1_2324/Game.cs:45]: Functionality for Multiple Rolls would be a good addition to increase the capability of the code.

File: CMP1903_A1_2324/Game.cs [hunk 12]
```diff
@@ -15,7 +15,30 @@ internal class Game
          * rolls could be continous, and the totals and other statistics could be summarised for example.
          */
 
-        //Methods
+        /// <summary>  Creates three die objects, rolls them and outputs the values and their sum. </summary>
+        /// 
+        /// <returns> first die value, second die value, third die value, sum of all dice. </returns>
```
Comment (Tr0-p) [inline CMP1903_A1_2324/Game.cs]: Good use of XML commenting. Returns should be inside the function as it is not run upon instantiation.

File: CMP1903_A1_2324/Program.cs (hunk 5 above)
Comment (Tr0-p) [inline CMP1903_A1_2324/Program.cs]: A use of a indefinite while loop may be a better choice compared to a definite for loop.

File: CMP1903_A1_2324/Testing.cs (hunk 4 above)
Comment (Tr0-p) [reply CMP1903_A1_2324/Testing.cs:37]: I agree with Jack, the use of and will decrease the amount of repeated lines.

**Review by Tr0-p — COMMENTED** [review]
Only a few commenting errors which can be fixed alongside some functionality clean-ups, overall a good piece of code!

1. Have the requirements been met? Yes
2. Is the code formatted using the Style Guidelines correctly? Yes
3. Is the code easy to read? Yes
4. Are different errors handled correctly? There is no evidence of error handling pres

File: CMP1903_A1_2324/Testing.cs (hunk 4 above)
Comment (morganblackwell) [reply CMP1903_A1_2324/Testing.cs:37]: Decided to keep these lines separate to provide more information about the errors

**Pull request merged**